src/auth/index.ts

```diff
@@ -1,3 +1,46 @@
+import { ConsoleClient, TokenRequest, TokenResponse } from "../satellite"
+import { ElectricConfig } from "../satellite/config"
+import { fetch } from "cross-fetch"
+import Log from 'loglevel'
+
 export interface AuthState {
-  // ...
+  app: string,
+  env: string,
+  clientId: string
+  token?: string
+  refreshToken?: string
 }
+
+export class ConsoleHttpClient implements ConsoleClient {
+
+  config: ElectricConfig
+
+  constructor(config: ElectricConfig) {
+    this.config = config
+
+    if (!!config.console?.host) {
+      // we always set the default, if not set it's an error
+      throw Error("config.console must be set")
+    }
+  }
+
+  async token({ app, env, clientId }: TokenRequest): Promise<TokenResponse> {
```

Comment:
Minor note but it seems to me that `token` is a bit inscrutable. This is a console client. Which token is this? Maybe calling it `authToken` would convey more?

Also, does the signature and implementation here require `app` and `env`. Or is it OK to send e.g.: `env: null` in the post data? Also, why is `username` `cientId`? Is it not possible that the username would be a real username? And should we capture here that this is currently an insecure implementation?

Is this method exposed/intended for the end user to call?

Reply:
I think these questions should be answered in the context of implementing real auth. For now, this is just making the http request for getting the 'token' reusable.

The console client is internal only, so we can change things anytime. I currently don't know the scope of the functionality of the console that we want to expose to the client to answer if this should be made a first-class component.

Reply:
requesting the token is done as part of 'start replication' procedure. The model in my head is that even if you wait on electrify, this will not block on start().

Reply:
👌